Added eachrow/col/slice for v1.0 (#658)
JuliaLang/Compat.jl: README.md

## New functions, macros, and methods

* `eachslice` to iterate over some dimension of a matrix ([#29749]).

* `eachcol` to iterate over the second dimension of a matrix ([#29749]).

> * `eachrow` to iterate over the first dimension of a matrix ([#29749]).

Reviewer: Maybe consolidate these into a single item?

## New functions, macros, and methods

> * `eachcol` to iterate over the second dimension of a matrix and `eachrow` to iterate over the first dimension of a matrix and `eachslice` to iterate over some dimension of a matrix ([#29749]).

Reviewer: Maybe more compactly like so?
```suggestion
* `eachrow`, `eachcol`, and `eachslice` to iterate over first, second, or given dimension
  of an array ([#29749]).
```
No that it matters much...
